code-challenges.test.js

```diff
@@ -13,40 +15,135 @@
 // --------------------1) Create a function that takes in an array of objects and returns an array with a sentence about each person with their name capitalized.
 
 // a) Create a test with an expect statement using the variable provided.
-
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe("capitalizeName", () => {
+  it("returns an array with a sentence about each person with their name capitalized.", () => {
+      const hitchhikersCharacters = [
+          { name: "ford prefect", occupation: "a hitchhiker" },
+          { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+          { name: "arthur dent", occupation: "a radio employee" }
+        ]
+        // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+      expect(capitalizeName(hitchhikersCharacters)).toEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+  })
+})
 
 // b) Create the function that makes the test pass.
+const capitalizeName = (nestedObj) => {
+  //since the output needs to be a new array with the similar values, a high order map method is needed 
+   return nestedObj.map(value => {
+       let seperateStr = value.name.split(" ")
+       console.log(seperateStr)
+       let capFirstStrChar = seperateStr[0][0].toUpperCase() + seperateStr[0].slice(1);
+       //separateStr[0][0] accesses the first character of the first word. The second bracket notation that is concatinated will slice through the string to make sure 
+       //the full word itself is printed out, not just the first character itself, for instance, it will print "Ford" sucessfully instead of just "F."
+       console.log(capFirstStrChar)
+        let capSecondStrChar = seperateStr[1][0].toUpperCase() + seperateStr[1].slice(1)
```

Comment:
Careful when you start targeting specific indices. For example, what if someone had a 2 word first name? The code works for the given example variable, I'm just commenting this for you to keep in mind as you move forward 🙂